pkg/dependency/parser/python/pylock/parse.go

```diff
@@ -0,0 +1,64 @@
+package pylock
+
+import (
+	"github.com/BurntSushi/toml"
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/trivy/pkg/dependency"
+	"github.com/aquasecurity/trivy/pkg/dependency/parser/python"
+	ftypes "github.com/aquasecurity/trivy/pkg/fanal/types"
+	"github.com/aquasecurity/trivy/pkg/log"
+	xio "github.com/aquasecurity/trivy/pkg/x/io"
+)
+
+type Pylock struct {
+	Packages []Package `toml:"packages"`
+}
+
+type Package struct {
+	Name           string       `toml:"name"`
+	Version        string       `toml:"version"`
+	RequiresPython string       `toml:"requires-python"`
+	Dependencies   []Dependency `toml:"dependencies"`
+}
+
+type Dependency struct {
+	Name   string `toml:"name"`
+	Marker string `toml:"marker"`
+}
+
+// Parser parses pylock.toml defined in PEP 751.
+// https://peps.python.org/pep-0751
+type Parser struct {
+	logger *log.Logger
+}
+
+func NewParser() *Parser {
+	return &Parser{
+		logger: log.WithPrefix("pylock"),
+	}
+}
+
+func (p *Parser) Parse(r xio.ReadSeekerAt) ([]ftypes.Package, []ftypes.Dependency, error) {
+	var lock Pylock
+	if _, err := toml.NewDecoder(r).Decode(&lock); err != nil {
+		return nil, nil, xerrors.Errorf("failed to decode pylock.toml: %w", err)
+	}
+
+	var pkgs []ftypes.Package
+	var deps []ftypes.Dependency
```

Comment:
Can we also analyze dependency relationships?

Reply:
>I tried setting up a pylock.toml using pip, uv and poetry and in all cases, the pylock.toml had no dependencies.

Do you mean this field is always empty?
https://peps.python.org/pep-0751/#packages-dependencies